src/Widgets/ToolbarButton.vala

```diff
@@ -100,15 +99,67 @@ public class ENotes.ToolbarButton : Gtk.Button {
                 } else if (type == 2) {
                     plugin.request_string ("");
                 } else {
-                    code_buffer.insert_at_cursor (first_half, -1);
-                    int end_first_half_pos = code_buffer.cursor_position;
-                    code_buffer.insert_at_cursor (second_half, -1);
-
                     Gtk.TextIter cursor_position;
-                    code_buffer.get_iter_at_offset (out cursor_position, end_first_half_pos);
-                    code_buffer.place_cursor (cursor_position);
+                    code_buffer.get_iter_at_offset (out cursor_position, code_buffer.cursor_position);
+
+                    if (cursor_position.inside_word () || cursor_position.ends_word ()) {
+                        // gets word the cursor is currently on and modify it
+                        start = end = cursor_position;
+                        identify_word (ref start, ref end, first_half, second_half);
+                        var text = start.get_text (end);
+                        var wrapped_text = WordWrapper.get_wrapped_text (text, first_half, second_half);
+                        changes_text (ref code_buffer, start, end, wrapped_text);
+                    } else {
+                        // prints the wrapping text and put cursor in the middle 
+                        code_buffer.insert_at_cursor (first_half + second_half, -1);
+                        code_buffer.get_iter_at_offset (out cursor_position, code_buffer.cursor_position);
+                        cursor_position.backward_chars (second_half.length);
+                        code_buffer.place_cursor (cursor_position);
+                    }
                 }
             }
         });
     }
+
+    /**
```

Comment:
I think all this code bellow should maybe live inside the word wrapper class, or what do you think? :) (At least, it doesn't sound like something ToolbarButton should take care of) 

Maybe just identify_word and the other 2 static functions

Reply:
Completely agree, just wanted your feedback before applying these changes

Reply:
Yeah, maybe it will work better like that now that you've added all this functionality :)

Reply:
But leave that for a separate PR